lib/rubocop/cop/lint/inherit_exception.rb

```diff
@@ -0,0 +1,70 @@
+# encoding: utf-8
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop looks for error classes inheriting from `Exception`.
+      # It is configurable to suggest using either `RuntimeError` (default) or
+      # `StandardError instead.`
+      #
+      # @example
+      #
+      #   # bad
+      #
+      #   class C < Exception; end
+      #
+      # @example
+      #
+      #   # EnforcedStyle: runtime_error (default)
+      #
+      #   # good
+      #
+      #   class C < RuntimeError; end
+      #
+      # @example
+      #
+      #   # EnforcedStyle: standard_error
+      #
+      #   # good
+      #
+      #   class C < StandardError; end
+      class InheritException < Cop
+        include ConfigurableEnforcedStyle
+
+        MSG = 'Avoid inheriting from `Exception`. Instead use `%s`.'.freeze
+        PREFERRED_BASE_CLASS = {
+          style: {
+            runtime_error: 'RuntimeError',
+            standard_error: 'StandardError'
+          }
+        }.freeze
+
+        def on_class(node)
+          _class, base_class, _body = *node
+
+          return if base_class.nil?
+
+          check(base_class)
+        end
+
+        private
+
+        def check(node)
+          add_offense(node, :expression, format(MSG, preferred_class)) if
```

Comment:
Only use modifier `if` when the entire expression fits on one line. (We should have a cop for that.)